rtos/MemoryPool.h

```diff
@@ -184,13 +184,11 @@ class MemoryPool : private mbed::NonCopyable<MemoryPool<T, pool_sz> > {
 private:
     osMemoryPoolId_t             _id;
     /* osMemoryPoolNew requires that pool block size is a multiple of 4 bytes. */
-    char                         _pool_mem[((sizeof(T) + 3) & ~3) * pool_sz];
+    char                         _pool_mem[MBED_RTOS_STORAGE_MEM_POOL_MEM_SIZE(pool_sz, sizeof(T))];
```

Comment:
The change leaves the comment above unnecessary.

I do worry a bit about alignment, but maybe some C++11 `alignas` qualifier(s) could be added later.

Reply:
https://github.com/ARMmbed/mbed-os/pull/10706/commits/e621e810f6081b91fe59debedf982bbd2e7c81a0